### PR TITLE
Alerting: Add warning in telegram contact point

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

Telegram messages have [a size limit of 4096 UTF-8 characters](https://limits.tginfo.me/en). 

Because of this issue, messages above this size are not sent. We are working on a proper solution to truncate these messages , but [it might be that even truncating them](https://github.com/grafana/grafana/issues/89393), we get an error (because of the bad HTML format) and the alerts are still not sent.

In the meantime, we need users to be aware of this limitation.
This PR adds a warning in the UI when creating/editing a telegram contact point. It also adds a note in the documentation.

**Why do we need this feature?**

Some alert notifications can be lost. 

**Who is this feature for?**

All users using Telegram contact points

**Special notes for your reviewer:**

Screenshot 2024-06-19 at 16 14 24
Screenshot 2024-06-19 at 16 14 11

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by brendamuir — APPROVED** [review]

**Comment by gillesdemey** [comment]
Telegram is not the only one that limits the size of the payload – and truncating seems sensible. I think the main problem is that the request gets _rejected_ when using `parse_mode` is something other than `None` and the truncation creates invalid HTML (ie. not alerts are being delivered, not even in a truncated form).

This is arguably worse, I think we should only show the warning if the parse mode is not `None` for now and explain that the message may not get delivered at all.

File: public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx [hunk 1]
```diff
@@ -53,6 +53,7 @@ export function ChannelSubForm<R extends ChannelValues>({
 
   const { control, watch, register, trigger, formState, setValue } = useFormContext();
   const selectedType = watch(fieldName('type')) ?? defaultValues.type; // nope, setting "default" does not work at all.
+  const parse_mode = watch(fieldName('settings.parse_mode')) ?? defaultValues.settings.parse_mode;
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx:56]: Can `settings` be an empty object? I don't think we get type hinting here (ie. everything is `any`, right?)

Should we use optional chaining?

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx:56]: good point

File: public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx [hunk 2]
```diff
@@ -53,6 +53,7 @@ export function ChannelSubForm<R extends ChannelValues>({
 
   const { control, watch, register, trigger, formState, setValue } = useFormContext();
   const selectedType = watch(fieldName('type')) ?? defaultValues.type; // nope, setting "default" does not work at all.
+  const parse_mode = watch(fieldName('settings.parse_mode')) ?? defaultValues.settings?.parse_mode;
```
Comment (tomratcliffe) [inline public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx:56]: The defaultValue of `parse_mode` is an object, isn't it? So this should be `parse_mode.value`?

File: docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-telegram.md [hunk 3]
```diff
@@ -22,6 +22,11 @@ Use the Grafana Alerting - Telegram integration to send [Telegram](https://teleg
 
 ## Before you begin
 
+### Telegram limitation
+
+Telegram messages are limited to 4096 UTF-8 characters. If you use a `parse_mode` other than `None`, truncation might result in an invalid message, and causing the notification to fail.
```
Comment (brendamuir) [inline docs/sources/alerting/configure-notifications/manage-contact-points/integrations/configure-telegram.md:27]:
```suggestion
Telegram messages are limited to 4096 UTF-8 characters. If you use a `parse_mode` other than `None`, truncation may result in an invalid message, causing the notification to fail.
```

File: public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx (hunk 2 above)
Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/receivers/form/ChannelSubForm.tsx:56]: actually, when creating new contact point for telegram, in `defaultValues.settings.parse_mode` we have `undefined` (I've jsut checked). So I will remove the default value and change it only with  `const parse_mode = watch(fieldName('settings.parse_mode'))`

**Comment by brendamuir** [comment]
lgtm from docs!

**Review by gillesdemey — APPROVED** [review]

**Review by brendamuir — APPROVED** [review]

**Pull request merged**